File: Problems/217 Contains Duplicate/Solution.py
# https://leetcode.com/problems/contains-duplicate/
"""
Runtime: 488 ms, faster than 87.98% of Python3 online submissions for Contains Duplicate.
Memory Usage: 26.1 MB, less than 28.80% of Python3 online submissions for Contains Duplicate.
Next challenges:
"""


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(set(nums)) != len(nums)

File: Problems/217 Contains Duplicate/Solution.ts
// https://leetcode.com/problems/contains-duplicate/

/*
Runtime: 105 ms, faster than 86.59% of TypeScript online submissions for Contains Duplicate.
Memory Usage: 51.4 MB, less than 44.93% of TypeScript online submissions for Contains Duplicate.
*/
function containsDuplicate(nums: number[]): boolean {
    return new Set(nums).size != nums.length;
};

console.log(containsDuplicate([1, 2, 3, 1])); //true
console.log(containsDuplicate([1, 2, 3, 4])); //false
console.log(containsDuplicate([1, 1, 1, 3, 3, 4, 3, 2, 4, 2])); //true